File: pset6/credit.py
from cs50 import get_string

s = get_string("Number: ")

total = 0

for i in range(len(s) - 2, -1, -2):
    mul = (2 * int(s[i]))
    while (mul):
        total += mul % 10
        mul = int(mul / 10)


for i in range(len(s) - 1, -1, -2):
    total += int(s[i])

if total % 10 != 0:
    print("INVALID")
else:
    if len(s) == 15 and s[:2] in ['34', '37']:
        print("AMEX")
    elif len(s) == 16 and s[:2] in ['51', '52', '53', '54', '55']:
        print("MASTERCARD")
    elif (len(s) == 13 or len (s) == 16) and s[0] == '4':
        print("VISA")
    else:
        print("INVALID")

File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main (void)
{
    long number = get_long("Number: ");
    int count = 1;
    int sum1 = 0;
    int sum2 = 0;
    int sum = 0;
    int digitNumber;
    int digitNumber2;
    int muldigit;
    
    while(number)
    {
        digitNumber2 = digitNumber;
        
        digitNumber = number % 10;
        number /= 10;
        
        if (count % 2 != 0)
        {
            sum1 += digitNumber;
        }
        else
        {
            muldigit = digitNumber * 2;
            
            while (muldigit)
            {
                sum2 += muldigit % 10;
                muldigit /= 10; 
            }
        }
        count += 1;

    }
    sum = sum1 + sum2;
    
    if (sum % 10 != 0)
    {
        printf("INVALID\n");
    }
    else
    {
        if (count == 16 && digitNumber == 3)
        {
            if (digitNumber2 == 4 || digitNumber2 == 7)
            {
                printf("AMEX\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if (count == 17 && digitNumber == 5)
        {
            if (digitNumber2 > 0 && digitNumber2 < 6)
            {
                printf("MASTERCARD\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if ((count == 17 || count == 14) && digitNumber == 4)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    
}